File: client-side/src/app/core/services/parkingreservation.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import {environment} from "../../../environments/environment.development";

@Injectable({
  providedIn: 'root',
})
export class ParkingReservationService {
  url: string;

  constructor(private httpClient: HttpClient) {
    this.url =  `${environment.gateway_API}/parking/api`;
  }

  addParkingReservation(reservation: any) {
    return this.httpClient.post(`${this.url}/parking-reservation`, reservation);
  }

  updateParkingReservation(id: string, reservation: any) {
    return this.httpClient.put(
      `${this.url}/parking-reservation/${id}`,
      reservation
    );
  }

  getAllParkingReservations() {
    return this.httpClient.get(`${this.url}/parking-reservation/all`);
  }

  getParkingReservation(id: string) {
    return this.httpClient.get(`${this.url}/parking-reservation/${id}`);
  }

  deleteParkingReservation(id: string) {
    return this.httpClient.delete(`${this.url}/parking-reservation/${id}`);
  }

  confirmParkingReservation(id: string) {
    return this.httpClient.put(
      `${this.url}/parking-reservation/${id}/confirm`,
      {}
    );
  }

  declineParkingReservation(id: string) {
    return this.httpClient.put(
      `${this.url}/parking-reservation/${id}/decline`,
      {}
    );
  }

  getReservationsForClient() {
    return this.httpClient.get(`${this.url}/client-reservations`);
  }
}

File: ms_parking_management/src/main/java/com/example/ms_parking_management/services/interfaces/ParkingReservationService.java
package com.example.ms_parking_management.services.interfaces;

import java.util.List;
import java.util.Optional;

import com.example.ms_parking_management.entities.ParkingReservation;
import com.example.ms_parking_management.entities.ParkingSpace;

public interface ParkingReservationService {
	ParkingReservation addParkingReservation(ParkingReservation parkingReservation);

    Optional<ParkingReservation> getParkingReservation(String id);

    List<ParkingReservation> getAllParkingReservations();

    boolean deleteParkingReservation(String id);
    
    void confirmParkingReservation(String id);
    
    void declineParkingReservation(String id);
    
  

}
